lib/datadog/tracing/contrib/graphql/unified_trace.rb

```diff
@@ -144,11 +165,17 @@ def trace(callable, trace_key, resource, **kwargs)
                 Contrib::Analytics.set_sample_rate(span, config[:analytics_sample_rate])
               end
 
-              yield(span) if block_given?
+              if (before_callable = before || before_block)
+                before_callable.call(span)
+              end
```

Comment:
Worth perhaps raising if there's a `before && before_block`, just to make sure we don't accidentally introduce bugs in the future?

Reply:
👍